<a href="https://colab.research.google.com/github/adnanfathoni88/Big-Data/blob/main/03_Korelasi_regresi_berganda.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Import Dataset

In [7]:
# import library
import pandas as pd
import numpy as np

df = pd.read_csv('/content/drive/MyDrive/KULIAH/2024/BIG DATA/dataset/Global_AI_Content_Impact_Dataset.csv');
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 12 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   Country                                     200 non-null    object 
 1   Year                                        200 non-null    int64  
 2   Industry                                    200 non-null    object 
 3   AI Adoption Rate (%)                        200 non-null    float64
 4   AI-Generated Content Volume (TBs per year)  200 non-null    float64
 5   Job Loss Due to AI (%)                      200 non-null    float64
 6   Revenue Increase Due to AI (%)              200 non-null    float64
 7   Human-AI Collaboration Rate (%)             200 non-null    float64
 8   Top AI Tools Used                           200 non-null    object 
 9   Regulation Status                           200 non-null    object 
 10  Consumer Trust

# korelasi


In [19]:
numeric_df = df.select_dtypes(include=np.number)
correlation_matrix = numeric_df.corr()

correlation_matrix.corr()

Year  AI Adoption Rate (%)  \
Year                                        1.000000             -0.064083   
AI Adoption Rate (%)                       -0.064083              1.000000   
AI-Generated Content Volume (TBs per year) -0.099656             -0.252672   
Job Loss Due to AI (%)                     -0.423280             -0.200810   
Revenue Increase Due to AI (%)             -0.371465             -0.168401   
Human-AI Collaboration Rate (%)            -0.022788             -0.056166   
Consumer Trust in AI (%)                   -0.192852             -0.391445   
Market Share of AI Companies (%)           -0.050129              0.000767   

                                            AI-Generated Content Volume (TBs per year)  \
Year                                                                         -0.099656   
AI Adoption Rate (%)                                                         -0.252672   
AI-Generated Content Volume (TBs per year)                                    1.000000   
Job Loss Due to AI (%)                                                       -0.057816   
Revenue Increase Due to AI (%)                                               -0.114388   
Human-AI Collaboration Rate (%)                                              -0.144851   
Consumer Trust in AI (%)                                                     -0.075119   
Market Share of AI Companies (%)                                             -0.236348   

                                            Job Loss Due to AI (%)  \
Year                                                     -0.423280   
AI Adoption Rate (%)                                     -0.200810   
AI-Generated Content Volume (TBs per year)               -0.057816   
Job Loss Due to AI (%)                                    1.000000   
Revenue Increase Due to AI (%)                            0.148334   
Human-AI Collaboration Rate (%)                          -0.174634   
Consumer Trust in AI (%)                                  0.017040   
Market Share of AI Companies (%)                         -0.034707   

                                            Revenue Increase Due to AI (%)  \
Year                                                             -0.371465   
AI Adoption Rate (%)                                             -0.168401   
AI-Generated Content Volume (TBs per year)                       -0.114388   
Job Loss Due to AI (%)                                            0.148334   
Revenue Increase Due to AI (%)                                    1.000000   
Human-AI Collaboration Rate (%)                                  -0.025645   
Consumer Trust in AI (%)                                         -0.085384   
Market Share of AI Companies (%)                                 -0.174697   

                                            Human-AI Collaboration Rate (%)  \
Year                                                              -0.022788   
AI Adoption Rate (%)                                              -0.056166   
AI-Generated Content Volume (TBs per year)                        -0.144851   
Job Loss Due to AI (%)                                            -0.174634   
Revenue Increase Due to AI (%)                                    -0.025645   
Human-AI Collaboration Rate (%)                                    1.000000   
Consumer Trust in AI (%)                                          -0.160478   
Market Share of AI Companies (%)                                  -0.279838   

                                            Consumer Trust in AI (%)  \
Year                                                       -0.192852   
AI Adoption Rate (%)                                       -0.391445   
AI-Generated Content Volume (TBs per year)                 -0.075119   
Job Loss Due to AI (%)                                      0.017040   
Revenue Increase Due to AI (%)                             -0.085384   
Human-AI Collaboration Rate (%)                         

Regresi Linear Berganda

In [17]:
# multiple linear regression

np.random.seed(42)
split = np.random.rand(len(df)) < 0.8
df_train = df[split]
df_test = df[~split]

# data train
x_train = np.asanyarray(df_train[['Human-AI Collaboration Rate (%)', 'AI Adoption Rate (%)', 'Revenue Increase Due to AI (%)', 'Consumer Trust in AI (%)' ]])
y_train = np.asanyarray(df_train[['Job Loss Due to AI (%)']])

# data test
x_test = np.asanyarray(df_test[['Human-AI Collaboration Rate (%)', 'AI Adoption Rate (%)', 'Revenue Increase Due to AI (%)', 'Consumer Trust in AI (%)' ]])
y_test = np.asanyarray(df_test[['Job Loss Due to AI (%)']])

In [20]:
from sklearn.linear_model import LinearRegression

# membuat model dan melatih model
lr_model = LinearRegression()
lr_model.fit(x_train, y_train)

# menampilkan koefisien dan intercept
print('Koefisien: ', lr_model.coef_)
print('Intercept / konstanta: ', lr_model.intercept_)

Koefisien:  [[0.06598105 0.00103899 0.08825793 0.07951572]]
Intercept / konstanta:  [13.5351115]


Job Loss Due to AI (%) = 13.5351 + (0.0660 * Human-AI Collaboration Rate) + (0.0010 * AI Adoption Rate) + (0.0883 * Revenue Increase Due to AI) + (0.0795 * Consumer Trust in AI)

Angka-angka ini (koefisien dan intercept) adalah parameter yang telah "dipelajari" oleh model dari data pelatihan dan akan digunakan untuk membuat prediksi tentang Job Loss Due to AI (%) berdasarkan nilai dari empat variabel independen tersebut.